File: backend/task-service/src/main/java/org/driventask/task/Payload/Response/TaskResponse.java
package org.driventask.task.Payload.Response;

import java.time.LocalDateTime;
import java.util.Set;

import org.driventask.task.Enum.EPriority;
import org.driventask.task.Enum.EStatus;

public record TaskResponse(
    String id,

    String title,

    String description,

    LocalDateTime dueDate,

    EPriority priority,

    EStatus status,
    
    Set<String> historyOfTask,

    Set<String> filesIncluded,

    String userAssigned,

    String projectId,

    LocalDateTime createdAt,

    LocalDateTime updatedAt

) {
    
}

File: frontend2/src/app/models/task-response.ts
import { FileResponse } from "./file-response";
import { EPriority } from "./priority";
import { EStatus } from "./status";

export interface TaskResponse {
    id: String;
    title: String;
    description: String;
    dueDate: Date;
    priority: EPriority;
    status: EStatus;
    historyOfTask: String[];
    filesIncluded: String[];
    files?: FileResponse[];
    userAssigned: String;
    projectId: String;
    createdAt: Date;
    updatedAt: Date;
}
